### PR TITLE
EditorInput - remove auto scroll on content change

### PR DESCRIPTION
**Pull request by sra405**
Fixes - https://raspberrypifoundation.slack.com/archives/C022FDFRM7B/p1730800544670889

### CODE REVIEW

File: src/components/Editor/EditorInput/EditorInput.jsx [hunk 1]
```diff
@@ -95,19 +97,26 @@ const EditorInput = () => {
   }, [project?.components]);
 
   useEffect(() => {
-    if (!project?.components) return;
+    const newFileNames = project.components.map(
+      (file) => `${file.name}.${file.extension}`,
+    );
+    if (newFileNames.join() !== fileNames?.join()) {
```
Comment (conorriches) [inline src/components/Editor/EditorInput/EditorInput.jsx:103]: Just checking - things won't be in a different order right?

**Review by conorriches — COMMENTED** [review]
Nice! Quick question that's all

File: src/components/Editor/EditorInput/EditorInput.jsx (hunk 1 above)
Comment (sra405) [reply src/components/Editor/EditorInput/EditorInput.jsx:103]: nice thinking and challenge. Just checking but it shouldn't matter. The only way we used to be able to alter the file order is through drag/drop. But the features currently not working and even if it was you'd probably want to scroll then anyway. This current approach scrolls for opened/reopened files but doesn't scroll for anything that doesn't change this array (i.e. `project.component.content` changes that currently invoke this on type)

Comment (conorriches) [reply src/components/Editor/EditorInput/EditorInput.jsx:103]: Roger! Makes sense

**Review by conorriches — APPROVED** [review]
Very nice!

**Pull request merged**